### PR TITLE
Changing IntegrationTestBase to use ApplicationId instead of Id.Application

### PR DESCRIPTION
**Pull request by CuriousVini**
Changing IntegrationTestBase to use ApplicationId instead of Id.Application

Issue: https://issues.cask.co/browse/CDAP-5690
Build: http://builds.cask.co/browse/CDAP-DUT4010-6

### CODE REVIEW

File: cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java [hunk 1]
```diff
@@ -179,7 +180,8 @@ public ApplicationManager deployApplication(Id.Application appId,
   }
 
   @Override
-  public ApplicationManager getApplicationManager(Id.Application appId) {
+  public ApplicationManager getApplicationManager(ApplicationId applicationId) {
+    Id.Application appId = Id.Application.from(applicationId.getNamespace(), applicationId.getApplication());
     return new RemoteApplicationManager(appId, clientConfig, restClient);
```
Comment (bdmogal) [inline cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java]: you can just do ``return new RemoteAppicationManager(applicationId.toId(), clientConfig, restClient)``

File: cdap-integration-test/src/test/java/co/cask/cdap/test/IntegrationTestBaseTest.java [hunk 2]
```diff
@@ -64,8 +65,8 @@ public void testDeployApplicationInNamespace() throws Exception {
   @Test
   public void testGetApplicationManager() throws Exception {
     ApplicationManager applicationManager = deployApplication(TestApplication.class);
-    ApplicationManager testApplicationManager = getApplicationManager(Id.Application.from(Id.Namespace.DEFAULT,
-                                                                                          TestApplication.NAME));
+    ApplicationId applicationId = new ApplicationId(Id.Namespace.DEFAULT.getId(), TestApplication.NAME);
```
Comment (bdmogal) [inline cdap-integration-test/src/test/java/co/cask/cdap/test/IntegrationTestBaseTest.java]: ``ApplicationId applicationId = NamespaceId.DEFAULT.app(TestApplication.NAME)``

File: cdap-unit-test/src/main/java/co/cask/cdap/test/UnitTestManager.java [hunk 3]
```diff
@@ -171,12 +173,13 @@ public ApplicationManager deployApplication(Id.Namespace namespace, Class<? exte
   public ApplicationManager deployApplication(Id.Application appId,
                                               AppRequest appRequest) throws Exception {
     appFabricClient.deployApplication(appId, appRequest);
-    return appManagerFactory.create(appId);
+    ApplicationId applicationId = new ApplicationId(appId.getNamespaceId(), appId.getId());
```
Comment (bdmogal) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/UnitTestManager.java]: ``appId.toEntityId()``

File: cdap-unit-test/src/main/java/co/cask/cdap/test/internal/ApplicationManagerFactory.java [hunk 4]
```diff
@@ -25,5 +25,5 @@
  */
 public interface ApplicationManagerFactory {
 
-  ApplicationManager create(@Assisted("applicationId") Id.Application applicationId);
+  ApplicationManager create(@Assisted("applicationId") ApplicationId applicationId);
```
Comment (bdmogal) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/internal/ApplicationManagerFactory.java]: the package name says ``internal``, but please confirm with @anwar6953 if it is ok to make this change

Comment (anwar6953) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/internal/ApplicationManagerFactory.java]: seems fine to me. Users can technically see this class, but can't (and shouldn't) do anything with it.

**Comment by CuriousVini** [comment]
@bdmogal Addressed comments. Please review when you get a chance.

**Comment by bdmogal** [comment]
LGTM. Squash and merge after a clean build

**Comment by CuriousVini** [comment]
Build is Green. Merging

**Pull request merged**